pkg/controller/component/component_version.go

```diff
@@ -275,7 +275,7 @@ func checkNMergeImages(serviceVersion string, appsInDef, appsInVer map[string]ap
 		if len(ver.name) == 0 {
 			match, err := CompareServiceVersion(serviceVersion, def.version)
 			if err != nil {
-				def.err = err
+				def.err = fmt.Errorf("failed to compare service version: %v", err)
```

Comment:
maybe use %w for the error and print more context (e.g. raw version strings).

Reply:
> maybe use %w for the error and print more context (e.g. raw version strings).

fixed